frankenphp.c

```diff
@@ -105,13 +106,23 @@ static uintptr_t frankenphp_clean_server_context() {
 
 static void frankenphp_request_reset() {
   zend_try {
-    int i;
+    frankenphp_server_context *ctx = SG(server_context);
 
-    for (i = 0; i < NUM_TRACK_VARS; i++) {
-      zval_ptr_dtor(&PG(http_globals)[i]);
-    }
+    if (ctx->worker_ready) {
+      int i;
+      for (i = 0; i < NUM_TRACK_VARS; i++) {
+        zval_ptr_dtor(&PG(http_globals)[i]);
+      }
 
-    memset(&PG(http_globals), 0, sizeof(zval) * NUM_TRACK_VARS);
+      // Restore worker script super globals
+      memcpy(&PG(http_globals), ctx->worker_http_globals,
```

Comment:
Pretty sure this will break GC since reference counting won't be performed and may either cause a memory leak or GC'ing of variables unexpectedly.

What you probably (maybe) want to do is create a hashmap copy of globals before starting the work, then after the work, restore that hashmap to globals